qiime/sdk/visualization.py

```diff
@@ -33,3 +35,14 @@ def _from_data_dir(cls, data_dir, provenance):
         # shutil.copytree doesn't allow the destination directory to exist.
         viz._archiver.save_data(data_dir, distutils.dir_util.copy_tree)
         return viz
+
+    @classmethod
+    def get_index_paths(cls, fp):
```

Comment:
I think this needs to be an instance method since `Visualization` can be constructed from a zipfile (`load`) or a directory (`_from_data_dir`). I think this method should exist on `Archiver` because it has access to its `self._data_dir`, which contains the index files and whatever else the visualizer produced. Note that `_data_dir` does not always exist, so you'll need to handle that with the following lines:

```python
if not os.path.exists(self._data_dir):
    self._extract_data()
```